### PR TITLE
Add overloaded loss functions

### PR DESCRIPTION
**Pull request by Shashi456**
Fix: #422 
Added overloaded functions for losses which have a reduction parameter. 
cc: @saeta, @eaplatanios, @rxwei.

Build passes locally.

One final question, @rxwei you mentioned that the reduction parameter in the original loss functions should no longer be made default, Should I go ahead and make those changes as well?

### CODE REVIEW

**Comment by Shashi456** [comment]
@rxwei done.

File: Sources/TensorFlow/BackwardsCompatibility.swift [hunk 1]
```diff
@@ -615,3 +618,138 @@ public extension Dense {
...
+///   - predicted: Predicted outputs from a neural network.
+///   - expected: Expected values, i.e. targets, that correspond to the correct output.
+@differentiable(wrt: predicted)
+public func l2Loss<Scalar: TensorFlowFloatingPoint>(
+    predicted: Tensor<Scalar>,
+    expected: Tensor<Scalar>
+) -> Tensor<Scalar> {
+    l2Loss(predicted: predicted, expected: expected, reduction: { $0.sum() })
+}
+
```
Comment (rxwei) [inline Sources/TensorFlow/BackwardsCompatibility.swift]: Where is `meanAbsoluteError`?

**Comment by rxwei** [comment]
Oh actually my bad. I didn't notice that they have no reduction parameter.

File: Sources/TensorFlow/BackwardsCompatibility.swift (hunk 1 above)
Comment (Shashi456) [reply Sources/TensorFlow/BackwardsCompatibility.swift]: @rxwei, these functions are already two parameter functions and don't need to be overloaded.

**Comment by rxwei** [comment]
> One final question, @rxwei you mentioned that the reduction parameter in the original loss functions should no longer be made default, Should I go ahead and make those changes as well?

There's no need to do that at the moment because there's no overload resolution ambiguity in the type checker.

**Comment by Shashi456** [comment]
Kokoro surprisingly doesn't seem to be getting triggered.

**Comment by Shashi456** [comment]
@rxwei this can be merged.
